components/definition/cellRules/NetPyNECellRules.js

```diff
@@ -361,179 +418,207 @@ export default class NetPyNECellRules extends React.Component {
+              iconStyle={{color: page != 'main' ? cyan500 : pink400, ...styles.home.icon}}
+              onClick={ () => this.setState({page: 'main', selectedCellRule: undefined, selectedSection: undefined, selectedMechanism: undefined})}
+            >
+              <HomeIcon/>
+            </IconButton>
+
+            <NavigationChevronRight style={styles.rightArrow}/>
+
+            <FloatingActionButton
+              zDepth={1}
+              id="newCellRuleButton"
+              style={styles.cellRule}
+              className={"actionButton smallActionButton"}
+              secondary={ page != 'main' }
+              data-tooltip={ 
+                page != 'main'
+                  ? selectedCellRule && selectedCellRule.length > 8 
+                    ? selectedCellRule 
+                    : "Go back to rule"
+                  : "Create rule"
                 }
-                anchorOrigin={{ horizontal: 'left', vertical: 'top' }}
-                targetOrigin={{ horizontal: 'left', vertical: 'top' }}
-              >
-              </IconMenu>
-            </div>
-            <div style={{ clear: "both" }}></div>
-            {sections}
-          </div>
-          <div className={"details"}>
-            {selectedSection}
-            {dialogPop}
-          </div>
-        </CardText>
-      );
+              onClick={() => this.handleHierarchyClick('main')}
+            >
+              { page != 'main' 
+                ? selectedCellRule.length > 8 
+                  ? (selectedCellRule.slice(0,7)+'...') 
+                  : selectedCellRule 
+                : <ContentAdd/>
+              }
+            </FloatingActionButton>
+            
+            <NavigationChevronRight style={styles.rightArrow}/>
+
+            <RaisedButton
+              id="newSectionButton"
+              style={styles.sections.container}
+              disabledBackgroundColor="grey"
+              buttonStyle={styles.sections.icon}
+              primary={ page != 'mechanisms' }
+              secondary={ page == 'mechanisms' }
+              disabled={ selectedCellRule == undefined }
+              onClick={ () => this.handleHierarchyClick('sections') }
+              data-tooltip={ page == 'mechanisms' 
+                ? !!selectedSection && selectedSection.length > 9 
+                  ? selectedSection 
+                  : "Go back to section"
+                : page == "sections" 
+                  ? "Create a section" 
+                  : !!selectedCellRule
+                    ?  !!model && !!model[selectedCellRule] && Object.keys(model[selectedCellRule]['secs']).length > 0
+                      ? "Explore sections"
+                      : "Create first section"
+                    : "No rule selected" 
+              }
+            >
+              <p style={{color: 'white', height: '100%'}}>
+                {page == "mechanisms" 
+                  ? selectedSection != undefined 
```

Comment:
Can we change this to a classic if/else? Probably you can even extract it to a method and maybe reuse in several places?